R/calculate_within_batch_ari.R

```diff
@@ -48,54 +48,61 @@ calculate_within_batch_ari <- function(individual_sce_list,
   }
 
   # Check that `pc_name` is in merged SCE object
-  if (!pc_name %in% reducedDimNames(merged_sce)) {
-    stop("The provided `pc_name` cannot be found in the `merged_sce`.")
+  if (!any(pc_list %in% reducedDimNames(merged_sce))) {
+    stop("One or more of the PC names provided in `pc_list` cannot be found in the `merged_sce`.")
   }
 
-  # Pull out the PCs or analogous reduction from merged object
-  merged_pcs <- reducedDim(merged_sce, pc_name)
+  # Define NULL object for final tibble
+  final_within_batch_ari_tibble <- NULL
 
-  # Cluster merged pcs only one time
-  merged_sce <- merged_sce |>
-    cluster_sce(
-      pca_name = pc_name,
-      BLUSPARAM = bluster::NNGraphParam(cluster.fun = "louvain", type = "jaccard"),
-      cluster_column_name = "merged_clusters"
-    )
+  for (pc_name in pc_list) {
+    # Pull out the PCs or analogous reduction from merged object
+    merged_pcs <- reducedDim(merged_sce, pc_name)
```

Comment:
So what I think we should do is take everything inside this `for` loop and put into a non-exported function that takes as input `merged_sce`, `pc_name`, and `batch_column`. 

Then you can easily change the `for` loop to a `purrr::map` call instead.

Reply:
Thanks for that tip @allyhawkins! I refactored things as you suggested above -- let me know if the changes look reasonable to you!!